pkg/pgclient/client.go

```diff
@@ -161,7 +166,7 @@ func (cfg *Config) GetNumConnections() (min int, max int, numCopiers int, err er
 
 	numCopiers = perProc * maxProcs
 	// we leave one connection per-core for non-copier usages
-	if numCopiers+maxProcs > max {
+	if numCopiers > max-maxProcs {
 		log.Warn("msg", fmt.Sprintf("had to reduce the number of copiers due to connection limits: wanted %v, reduced to %v", numCopiers, max/2))
 		numCopiers = max / 2
```

Comment:
shouldn't this just be max-maxProcs? rather than max/2?

Reply:
no because maxProcs may be > max. Changed comment.